src/coursescheduler/constraints.py

```diff
@@ -1,2 +1,164 @@
+#         self.courses_ = courses
+#
+#     def satisfied(self, assignment: Dict[Course, TimeSlotConfiguration]) -> bool:
+#         score = 0
+#         for course in self.courses_:
+#             if course not in assignment:
+#                 continue
+#             for timeslot in assignment[course].timeslots_:
+#                 for preferred_time in course.instructor_.preferred_times_:
+#                     if (timeslot.day_ == preferred_time.day_):
+#                         if (timeslot.start_ < preferred_time.start_):
+#                             score -= (preferred_time.start_ - timeslot.start_) ** 2
+#                         if (preferred_time.end_ < timeslot.end_):
+#                             score -= (timeslot.end_ - preferred_time.end_) ** 2
+#         if (score < -400):
+#             return False
+#
+#         return True
+#
+#
+# class InstructorNoDoubleBookingConstraint(Constraint[Course, TimeSlotConfiguration]):
+#     def __init__(self, courses: List[Course]) -> None:
+#         super().__init__(courses)
+#         self.courses_: List[Course] = courses
+#
+#     def satisfied(self, assignment: Dict[Course, TimeSlotConfiguration]) -> bool:
+#         timeslots = []
+#         for course in self.courses_:
+#             if course not in assignment:
+#                 continue
+#             timeslots += assignment[course].timeslots_
+#
+#         for i in range(len(timeslots)):
+#             for j in range(i + 1, len(timeslots)):
+#                 if (timeslots[i].day_ == timeslots[j].day_):
+#                     if (timeslots[i].start_ >= timeslots[j].start_ and timeslots[i].start_ <= timeslots[j].end_):
+#                         return False
+#                     if (timeslots[j].start_ >= timeslots[i].start_ and timeslots[j].start_ <= timeslots[i].end_):
+#                         return False
+#
+#         return True
+#
+#
+# class InstructorCoursePreferenceConstraint(Constraint[Course, Instructor]):
+#     def __init__(self, courses: List[Course]) -> None:
+#         super().__init__(courses)
+#         self.courses_: List[Course] = courses
+#
+#     # Constraint on per-professor means of preference scores
+#     def satisfied(self, assignment: Dict[Course, Instructor]) -> bool:
+#         assigned_courses_dict = {instructor.name_: [] for instructor in instructors}
+#         for course in self.courses_:
+#             if course not in assignment:
+#                 continue
+#             assigned_courses_dict[assignment[course].name_].append(course)
+#
+#         for instructor in instructors:
+#             if (len(assigned_courses_dict[instructor.name_]) > 0):
+#                 sum = 0
+#                 for course in assigned_courses_dict[instructor.name_]:
+#                     sum += instructor.qualifications_[course.name_]
+#                 mean = sum / len(assigned_courses_dict[instructor.name_])
+#                 if mean < 120:
+#                     return False
+#         return True
```

Comment:
It may be worth removing the commented soft constraints for now and leaving them for #35 and #14 as they may not even be used.

```suggestion
```